File: backend/app/schemas/resume.py
from datetime import datetime

from pydantic import BaseModel


class CreateResumeRequest(BaseModel):
    title: str


class UpdateResumeRequest(BaseModel):
    title: str | None = None
    template_id: str | None = None
    resume_data: dict | None = None


class UpdateTemplateRequest(BaseModel):
    template_id: str


class PersonalInfo(BaseModel):
    name: str = ""
    email: str = ""
    phone: str = ""
    location: str = ""
    linkedin: str = ""
    github: str = ""


class ExperienceEntry(BaseModel):
    company: str = ""
    position: str = ""
    description: str = ""


class ProjectEntry(BaseModel):
    title: str = ""
    description: str = ""


class EducationEntry(BaseModel):
    institution: str = ""
    degree: str = ""
    field: str = ""
    start_date: str = ""
    end_date: str = ""


class ResumeData(BaseModel):
    personal_info: PersonalInfo = PersonalInfo()
    summary: str = ""
    experience: list[ExperienceEntry] = []
    projects: list[ProjectEntry] = []
    skills: list[str] = []
    education: list[EducationEntry] = []
    certifications: list[dict] = []
    languages: list[dict] = []
    achievements: list[str] = []


class ResumeListResponse(BaseModel):
    id: str
    title: str
    target_role: str | None = None
    template: dict | None = None
    updated_at: datetime


class ResumeListOutput(BaseModel):
    items: list[ResumeListResponse]
    page: int
    limit: int
    total: int


class ResumeDetailResponse(BaseModel):
    id: str
    title: str
    template_id: str | None = None
    resume_data: dict
    ats_analysis: dict | None = None
    cover_letter: dict | None = None


class ResumeCreateResponse(BaseModel):
    id: str
    title: str


class MessageResponse(BaseModel):
    message: str

File: frontend/types/resume.ts
export interface ResumeListItem {
  id: string;
  title: string;
  target_role: string | null;
  template: { id: string; name: string } | null;
  updated_at: string;
}

export interface ResumeListResponse {
  items: ResumeListItem[];
  page: number;
  limit: number;
  total: number;
}

export interface ResumeDetail {
  id: string;
  title: string;
  template_id: string | null;
  resume_data: Record<string, unknown>;
  ats_analysis: Record<string, unknown> | null;
  cover_letter: Record<string, unknown> | null;
  target_role: string | null;
}

export interface CreateResumeRequest {
  title: string;
}

export interface CreateResumeResponse {
  id: string;
  title: string;
}

export interface PersonalInfo {
  name: string;
  email: string;
  phone: string;
  location: string;
  linkedin: string;
  github: string;
}

export interface ExperienceEntry {
  company: string;
  position: string;
  description: string;
  start_date: string;
  end_date: string;
}

export interface ProjectEntry {
  title: string;
  description: string;
  start_date: string;
  end_date: string;
}

export interface EducationEntry {
  institution: string;
  degree: string;
  field: string;
  start_date: string;
  end_date: string;
}

export interface ResumeData {
  personal_info: PersonalInfo;
  summary: string;
  experience: ExperienceEntry[];
  projects: ProjectEntry[];
  skills: string[];
  education: EducationEntry[];
}
